File: src/main/java/io/aif/associations/calculators/edge/PredefinedEdgeWeightCalculator.java
package io.aif.associations.calculators.edge;


import java.util.Map;

public class PredefinedEdgeWeightCalculator<T> implements IEdgeWeightCalculator<T> {
    
    private final Map<T, Map<T, Double>> connections;

    public PredefinedEdgeWeightCalculator(final Map<T, Map<T, Double>> connections) {
        this.connections = connections;
    }

    @Override
    public double calculate(final T from, final T to) {
        return connections.get(from).get(to);
    }
    
}

File: src/main/scala/io/aif/associations/calculators/edge/PredefinedEdgeWeightCalculator.scala
package io.aif.associations.calculators.edge

import io.aif.associations.model.graph.Graph


class PredefinedEdgeWeightCalculator[T](graph: Graph[T, Double]) {

  def calculate(from: T, to: T): Option[Double] = graph.edge(from, to)

}
